app/my-bets/page.tsx

```diff
@@ -7,96 +7,134 @@ import { Market, Position, tradesOutcomeBalance } from '@/entities';
 import { useAccount } from 'wagmi';
 
+export interface UserPositionComplete extends UserPosition {
+  fpmmTrades: FpmmTrade[];
+  market: FixedProductMarketMaker;
+  condition: Condition;
+}
+
 export default function MyBetsPage() {
   const { address } = useAccount();
-  const [unredeemedBets, setUnredeemedBets] = useState<UserPosition[]>([]);
-  const queryClient = useQueryClient();
 
-  const { data, isLoading } = useQuery({
-    queryKey: ['getUserPositions', address],
-    queryFn: () => getUserPositions({ id: address?.toLowerCase() as string }),
+  const { data: userPositionsComplete, isLoading } = useQuery<UserPositionComplete[]>({
+    queryKey: ['getUserPositionsComplete', address],
+    queryFn: async () => {
+      if (!address) return [];
+
+      const userPositionsData = await getUserPositions({ id: address.toLowerCase() });
+      const userPositions = userPositionsData?.userPositions ?? [];
+
+      const userPositionsComplete = await Promise.allSettled(
+        userPositions.map(
+          async (userPosition): Promise<UserPositionComplete | undefined> => {
+            try {
+              const position = new Position(userPosition.position);
+              const outcomeIndex = position.outcomeIndex - 1;
+
+              const conditionData = await getConditionMarket({
+                id: position.conditionId,
+              });
+              const condition = conditionData?.conditions[0];
+              const market =
+                condition && new Market(condition?.fixedProductMarketMakers[0]);
+
+              if (!market) return undefined;
+
+              const tradesData = await getMarketUserTrades({
+                creator: address.toLowerCase(),
+                fpmm: market.data.id,
+                outcomeIndex_in: [outcomeIndex],
+              });
+
+              return {
+                ...userPosition,
+                market: market.data,
+                condition,
+                fpmmTrades: tradesData?.fpmmTrades || [],
+              };
+            } catch (error) {
+              console.error(error);
+            }
+          }
+        )
+      ).then(results =>
+        results
+          .filter(
+            (result): result is PromiseFulfilledResult<UserPositionComplete> =>
+              result.status === 'fulfilled' && result.value !== undefined
+          )
+          .map(result => result.value)
+      );
+
+      return userPositionsComplete;
+    },
     enabled: !!address,
   });
 
-  const userPositions = data?.userPositions ?? [];
-
-  const filterActiveBets = userPositions.filter(
-    position => position.position.conditions[0].resolved === false
+  const sortBetsByNewest = (bets: UserPositionComplete[]) => {
+    return [...bets].sort((a, b) => {
+      return (
+        b.fpmmTrades[b.fpmmTrades.length - 1]?.creationTimestamp -
+        a.fpmmTrades[a.fpmmTrades.length - 1]?.creationTimestamp
+      );
+    });
+  };
```

Comment:
Do we need this function that is used only once?

Reply:
If we need it can be outside the component. Also we can sort inside the `useQuery`.

Reply:
I want this code with a description so it's clear on what it does.

Do you think a variable is better?

Reply:
I think a variable is better, as it's only used in one place as you mentioned. But I like it here. Since the useQuery is already doing too much.

Reply:
We can have it outside the component and sort them inside the `useQuery`. We just need what's inside sort.

Reply:
Updated it